File: queue.cpp

#include <iostream>
#include<conio.h>
using namespace std;
struct Node
{
int data;

Node *next;
};

Node *front = NULL;
Node *rear = NULL;
bool isempty()
{
 if(front == NULL && rear == NULL)
 return true;
 else
 return false;
}
void insertion ( )
{
    int val;
    cout<<"insert element:"<<endl;
    cin>>val;
 Node *ptr = new Node();
 ptr->data= val;
 ptr->next = NULL;
 if( front == NULL )
 {
  front = ptr;
  rear = ptr;
 }
 else
 {
  rear ->next = ptr;
  rear = ptr;
 }
}
void deletion ( )
{
 if( isempty() )
 cout<<"empty\n";
 else
 if( front == rear)
 {

  front = rear = NULL;
 }
 else
 {
  Node *ptr = front;
  front = front->next;

 }
}
void display()
{
 if (isempty())
  cout<<"empty\n";
 else
 {
  Node *ptr = front;
  while( ptr !=NULL)
  {
   cout<<ptr->data<<" ";
   ptr= ptr->next;
  }
 }
}
int main()
{
 int op;

 {
  cout<<"\n1.insertion";
  cout<<"\n2.deletion";
  cout<<"\n4.display";
  cout<<"\n5.edatait";
  do{
    cout<<"\nyour choice:"<<endl;
    cin>>op;

  switch (op)
  {
  case 1: cout<<insertion;

          break;
  case 2: deletion();
          break;
  case 3: display();
          break;
  case 4: cout<<"edatait"<<endl;
  break;
  default: cout<<"wrong choice"<<endl;
  }
 }
 while(op!=6);
 return 0;
 }
}

File: queue.py
class queuelist:
    def __init__(self, data, l=None, r=None):
        self.data = data
        self.link = None
 
 
class Queuelist:
    def __init__(self):
        self.rear = None
        self.front = None
        
    def deletion(self):          
        if self.front is None:
            print('Underflow')
            
 
        temp = self.front
        print('Removing…', temp.data)
 
        
        self.front = self.front.next
 
        
        if self.front is None:
            self.rear = None
 
        
        
 
        return temp.data
 
    def insertion(self, item):    
        node = queuelist(item)
        print('ADDING', item)
 
       
        if self.front is None:
            
            self.front = node
            self.rear = node
        else:
           
            self.rear.link = node
            self.rear = node
 
       
       
    
    def peek(self):
       
        if self.front:
            return self.front.x
        else:
            print("peek element is")
 
   
    def isEmpty(self):
        return self.rear is None and self.front is None
 
if __name__ == '__main__':
 
    q = Queuelist()
    q.insertion(11)
    q.insertion(22)
    q.insertion(33)
    q.insertion(44s)
 
    print('The  element is', q.peek())
 
    q.deletion()
    q.deletion()
    q.deletion()
    q.deletion()
 
    if q.isEmpty():
        print('Empty')
    else:
        print('Elements')
